### PR TITLE
Separating Test and Build tasks to two actions

### PR DESCRIPTION
**Pull request by quazirafi**
**PR Type**

- [ ] Breaking change
- [ ] Feature
- [x] Patch

**Brief Description**

This PR separates build and test tasks into two separate actions.

**Not In Scope**

Anything not related to separating build and test actions.

**PR Checklist**

- [x] Documentation (comments, API, user, design, and developer as applicable)
    - [API](https://nwchemex-project.github.io/.github/documenting/writing_api_documentation.html)
    - [User](https://nwchemex-project.github.io/.github/documenting/writing_tutorials.html)
    - [Design](https://nwchemex-project.github.io/.github/documenting/writing_design_documentation.html)
    - [Developer](https://nwchemex-project.github.io/.github/documenting/writing_developer_documentation.html)
- [x] Testing (unit, integration, performance, as applicable)
    - [Unit](https://nwchemex-project.github.io/.github/testing/unit.html)
    - [Integration](https://nwchemex-project.github.io/.github/testing/integration.html)
    - [Performance](https://nwchemex-project.github.io/.github/testing/performance.html)
- [x] Code adheres to organization standards
    - [C++](https://nwchemex-project.github.io/.github/coding/cxx/cxx_conventions.html)
    - [Python](https://nwchemex-project.github.io/.github/coding/python/python_conventions.html)
    - [Doxygen Conventions](https://nwchemex-project.github.io/.github/documenting/doxygen_conventions.html)
    - [reST Conventions](https://nwchemex-project.github.io/.github/documenting/rst_conventions.html)

**TODOs**

Nothing.

### CODE REVIEW

**Review by ryanmrichard — CHANGES_REQUESTED** [review]
Can you please:
- clean up the comments so they reflect what's going on instead of being copy/pasted and no longer relevant, 
- remove the stale commented out code (except the code coverage part, which needs to be re-enabled at some point), and 
- remove the binary file you checked in?

Thanks

File: actions/test/test.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+################################################################################
+#
+# The master build_and_test.sh script lives at
+# NWChemEx-Project/DeveloperTools/.github/workflows/scripts. The
+# build_and_test.sh script contained in all other NWChemEx-Project repositories
+# is a synchronized copy of the master script. Thus to make changes to the
+# build_and_test.sh file please make them to the master script.
+#
+################################################################################
```
Comment (ryanmrichard) [inline actions/test/test.sh:10]:
```suggestion
```

File: actions/build/build.sh [hunk 2]
```diff
@@ -22,8 +22,8 @@ set -e # Exit with error if any command fails
 
 arch=Linux-x86_64
 cmake_root=$(pwd)/cmake-"${cmake_version}"-"${arch}"
-cmake_command=cmake #"${cmake_root}/bin/cmake"
-ctest_command=ctest #"${cmake_root}/bin/ctest"
+cmake_command=cmake
+ctest_command=ctest
```
Comment (ryanmrichard) [inline actions/build/build.sh:26]:
```suggestion
```

**Review by ryanmrichard — CHANGES_REQUESTED** [review]
You still have the `.DS_store` binary thing, which shouldn't be checked in. 

Have you tested this PR and verified the tests run? Looking at `.github/workflows/c-cpp_tmpl.yaml` I don't see any calls to the testing action.

**Comment by quazirafi** [comment]
> You still have the `.DS_store` binary thing, which shouldn't be checked in.
> 
> Have you tested this PR and verified the tests run? Looking at `.github/workflows/c-cpp_tmpl.yaml` I don't see any calls to the testing action.

1. I removed that **.DS_store** file. In my second last push it is showing that the file is removed. 
2. Yes, I tested for ParallelZone and it worked. See here please:
https://github.com/NWChemEx-Project/ParallelZone/actions/runs/4160768371
@ryanmrichard

**Comment by ryanmrichard** [comment]
It's showing that it was renamed:

image

**Comment by quazirafi** [comment]
> It's showing that it was renamed:
> 
> image

Can you check now? @ryanmrichard

**Comment by quazirafi** [comment]
Hopefully, it should be fine now. I also removed the header comments. @ryanmrichard

**Comment by quazirafi** [comment]
Let me test one more time then we can merge this.

**Comment by quazirafi** [comment]
It looks fine to me now. I just tested with ParallelZone again with all changes. @ryanmrichard

File: actions/test/test.sh (hunk 1 above)
Comment (quazirafi) [reply actions/test/test.sh]: done

**Comment by ryanmrichard** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/NWChemEx-Project/.github/actions/runs/4205137489) **New version**:[v0.0.2](https://github.com/NWChemEx-Project/.github/releases/tag/v0.0.2) **Changes**:[v0.0.1...v0.0.2](https://github.com/NWChemEx-Project/.github/compare/v0.0.1...v0.0.2)

**Pull request merged**